misk-events-testing/src/main/kotlin/misk/events/FakeEventProcessor.kt

```diff
@@ -0,0 +1,106 @@
+package misk.events
+
+import java.util.concurrent.BlockingDeque
+import java.util.concurrent.LinkedBlockingDeque
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * An in-memory events system for testing publishers and consumers.
+ *
+ * To use this, first install the [FakeEventProcessorModule] in your program's test module:
+ *
+ * ```
+ * install(FakeEventProcessorModule)
+ * ```
+ *
+ * Next, use Guice multibindings to register consumers. You can use the same registrations for both
+ * test and production code.
+ *
+ * ```
+ * newMapBinder<Topic, Consumer.Handler>().addBinding(MY_TOPIC).to<MyConsumer>()
+ * ```
+ *
+ * To publish, inject a [Producer] and call [publish].
+ *
+ * To consume all published events, call [deliverAll]. All events will be delivered events to the
+ * corresponding consumers. Note that no call to [Consumer.subscribe] is necessary.
+ */
+@Singleton
+class FakeEventProcessor @Inject constructor(
+  private val consumers: Map<Topic, Consumer.Handler>
+) : Producer {
```

Comment:
We are removing the usages of the `misk-events` package and will deprecate it very soon. I would try not write new code on top of these abstractions

Reply:
What would you like to test in your service?

Reply:
Agree w/ @zhxnlai, I'd love if we didn't add anything else to the old events package so as to make the migration to eventing as smooth as possible — we're currently in the process of converting all producer and consumer apps.